pkg/keycloak/client.go

```diff
@@ -415,6 +568,12 @@ type KeycloakInterface interface {
 	DeleteClient(clientId, realmName string) error
 	UpdateClient(kcClient, objClient *v1alpha1.KeycloakClient, realmName string) error
 	ListClients(realmName string) ([]*v1alpha1.KeycloakClient, error)
+
+	CreateIdentityProvider(identityProvider *v1alpha1.KeycloakIdentityProvider, realmName string) error
+	GetIdentityProvider(alias, realmName string) (*v1alpha1.KeycloakIdentityProvider, error)
+	UpdateIdentityProvider(identityProvider *v1alpha1.KeycloakIdentityProvider, realmName string) error
+	DeleteIdentityProvider(alias, realmName string) error
+	ListIdentityProviders(realmName string) ([]*v1alpha1.KeycloakIdentityProvider, error)
```

Comment:
Refactoring of the kc client can be done later (think there might be a PR already started).